### PR TITLE
[tensorflow][test] Update TF data service test

### PR DESCRIPTION
**Pull request by Teng-xu**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Update tensorflow data service test and add distribute_epoch mode test for tf2.4

We want to disable dataservice test for 2.3 because the latest api does not work for 2.3 and dataservice is still experimental so we think we should focus on 2.4 right now since it is gonna keep updating.

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/container_tests/bin/testDataserviceDistribute [hunk 1]
```diff
@@ -0,0 +1,8 @@
+HOME_DIR=/test
+BIN_DIR=${HOME_DIR}/bin
+LOG_DIR=${HOME_DIR}/logs
+
+python ${BIN_DIR}/test_dataset_service.py --local=False --distribute=True|| exit 1
+
+exit 0
+
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/testDataserviceDistribute:8]: Please place `#!/bin/bash` at the beginning of the script to define how this script should executed, and insert `set -ex` in the beginning of the script to ensure that:
1. "-e": the script exits immediately if a command fails with a non-zero exit code
2. "-x": each command executed is printed to console

**Review by saimidu — COMMENTED** [review]
Please update the PR description to let anyone who looks at this PR in the future know that TF 2.3 no longer supports dataservice, which is why the test has been disabled for TF 2.3.

**Comment by Teng-xu** [comment]
> Please update the PR description to let anyone who looks at this PR in the future know that TF 2.3 no longer supports dataservice, which is why the test has been disabled for TF 2.3.

We want to disable dataservice test for 2.3 because the latest api does not work for 2.3 and dataservice is still experimental so we think we should focus on 2.4 right now since it is gonna keep updating.

File: test/dlc_tests/container_tests/bin/testDataserviceDistribute (hunk 1 above)
Comment (Teng-xu) [reply test/dlc_tests/container_tests/bin/testDataserviceDistribute:12]: Updated

Comment (saimidu) [reply test/dlc_tests/container_tests/bin/testDataserviceDistribute:12]: nit: You can now remove the `|| exit 1` on line 9.

File: test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py [hunk 2]
```diff
@@ -175,18 +176,19 @@ def test_tensorflow_addons_cpu(tensorflow_training, ec2_connection, tf2_only, cp
 # Helper function to test data service 
 def run_data_service_test(ec2_connection, tensorflow_training):
     ec2_connection.run('python3 -m pip install --upgrade pip')
-    ec2_connection.run('pip3 install tensorflow==2.3')
+    ec2_connection.run('pip3 install tensorflow==2.4.0rc3')
```
Comment (saimidu) [inline test/dlc_tests/ec2/tensorflow/training/test_tensorflow_training.py:179]: nit: Is it possible to change this from rc3 to `tensorflow==2.4`?

**Review by saimidu — APPROVED** [review]

**Pull request merged**